client/cpp/synnax/channel/channel.cpp

```diff
@@ -0,0 +1,123 @@
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "synnax/channel/channel.h"
+#include <grpcpp/grpcpp.h>
+#include "v1/channel.pb.h"
+#include "synnax/exceptions.h"
+#include "synnax/telem/telem.h"
+
+using namespace Synnax;
+
+std::string RETRIEVE_ENDPOINT = "/channel/retrieve";
+std::string CREATE_ENDPOINT = "/channel/create";
+
+
+Channel::Channel(const api::v1::Channel &ch) {
+    name = ch.name();
+    key = ch.key();
+    index = ch.index();
+    rate = Telem::Rate(ch.rate());
+    is_index = ch.is_index();
+    leaseholder = ch.leaseholder();
+    dataType = Telem::DataType(ch.data_type());
+}
+
+Channel::Channel(
+        std::string name,
+        Telem::DataType dataType,
+        Telem::Rate rate,
+        bool is_index,
+        std::uint32_t leaseholder,
+        ChannelKey index,
```

Comment:
Continuing from above, makes this redundant. Unless you want to have this additional ctor.